api/handlers/listhandler.py

```diff
@@ -394,4 +394,4 @@ def packfile(self, cont_name, **kwargs):
             else:
                 raise Exception('Not authorized')
 
-        return upload.process_upload(self.request, upload.Strategy.packfile)
+        return upload.process_upload(self.request, upload.Strategy.packfile, None, None, self.origin)
```

Comment:
That's a great point. I don't like doing stuff like this because the params inevitably change and I mess up :) Changing now.